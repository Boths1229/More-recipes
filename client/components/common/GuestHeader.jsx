import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @description functional component to render the header for unauthenticated user
 *
 * @returns {JSX} return JSX
 */
const GuestHeader = () => (
  <header>
    <nav className="navbar navbar-expand-lg navbar-light bg-green">
      <Link to="/" className="navbar-link nav-text" id="navlink">More-recipe</Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <Link className="navbar-link nav-text" id="navlink" to="/signup">Signup</Link>
          </li>
          <li className="nav-item">
            <Link className="navbar-link nav-text" id="navlink" to="/signin">Login</Link>
          </li>
        </ul>
      </div>
    </nav>
  </header>
);

export default GuestHeader;
