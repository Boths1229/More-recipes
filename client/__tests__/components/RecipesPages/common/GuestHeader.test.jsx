import React from 'react';
import { shallow } from 'enzyme';
import GuestHeader from '../../../../components/common/GuestHeader';

describe('Guest Header component', () => {
  it('matches snapshot', () => {
    const output = shallow(<GuestHeader />);
    expect(output).toMatchSnapshot();
  });
});
