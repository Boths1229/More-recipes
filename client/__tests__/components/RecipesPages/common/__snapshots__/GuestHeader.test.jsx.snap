// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guest Header component matches snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-green"
  >
    <Link
      className="navbar-link nav-text"
      id="navlink"
      replace={false}
      to="/"
    >
      More-recipe
    </Link>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <Link
            className="navbar-link nav-text"
            id="navlink"
            replace={false}
            to="/signup"
          >
            Signup
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="navbar-link nav-text"
            id="navlink"
            replace={false}
            to="/signin"
          >
            Login
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
